libr/reg/double.c

```diff
@@ -3,111 +3,324 @@
 #include <r_reg.h>
 #include <r_util.h>
 
-// TODO: add support for 80bit floating point value
+// Helper to select appropriate float profile based on register size and endianness
+static RCFloatProfile get_float_profile(RReg *reg, RRegItem *item, bool *success)
+{
+	const bool be = (reg->endian & R_SYS_ENDIAN_BIG) == R_SYS_ENDIAN_BIG;
+	RCFloatProfile profile;
 
-// long double = 128 bit
-R_API double r_reg_get_double(RReg *reg, RRegItem *item) {
-	R_RETURN_VAL_IF_FAIL (reg && item, 0.0f);
-	double vld = 0.0f;
-	double ret = 0.0f;
-	int off = BITS2BYTES (item->offset);
-	RRegSet *regset = &reg->regset[item->arena];
-	switch (item->size) {
+	if (success)
```

Comment:
run clang-format-radare2 in this file, we dont do c++ bracket placings in r2land